src/web/rustdoc.rs

```diff
@@ -736,14 +727,12 @@ pub(crate) async fn target_redirect_handler(
     mut conn: DbConnection,
     Extension(storage): Extension<Arc<AsyncStorage>>,
 ) -> AxumResult<impl IntoResponse> {
-    let version = match_version(&mut conn, &name, &req_version)
+    let matched_release = match_version(&mut conn, &name, &req_version)
         .await?
-        .into_canonical_req_version_or_else(|_| AxumNope::VersionNotFound)?
-        .into_version();
+        .into_canonical_req_version_or_else(|_| AxumNope::VersionNotFound)?;
 
-    let crate_details = CrateDetails::new(&mut conn, &name, &version, Some(req_version.clone()))
-        .await?
-        .ok_or(AxumNope::VersionNotFound)?;
+    let crate_details = CrateDetails::from_matched_release(&mut conn, &matched_release).await?;
+    let version = matched_release.into_version();
```

Comment:
We could avoid a bunch of cloning if `CrateDetails::from_matched_release` took the `MatchedRelease` by value, then expose `crate_details.version` to use here.